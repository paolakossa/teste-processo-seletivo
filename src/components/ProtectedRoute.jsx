import React from 'react';
import { useSelector } from 'react-redux';
import { Navigate } from 'react-router-dom';

export const ProtectedRoute = ({ children }) => {
  const token = useSelector((state) => state.userData.token);

  if (!token) {
    return <Navigate to="/auth" replace />;
  }

  return children;
};
