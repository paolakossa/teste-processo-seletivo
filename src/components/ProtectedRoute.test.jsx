import React from 'react';
import { render, screen } from '@testing-library/react';
import { expect, describe, it } from 'vitest';
import { MemoryRouter, Routes, Route } from 'react-router-dom';
import { Provider } from 'react-redux';
import { configureStore } from '@reduxjs/toolkit';
import { ProtectedRoute } from './ProtectedRoute';

const renderWithRouterAndStore = (initialEntries, preloadedState) => {
  const store = configureStore({
    reducer: {
      userData: (state = preloadedState.userData) => state,
    },
  });

  return render(
    <Provider store={store}>
      <MemoryRouter initialEntries={initialEntries}>
        <Routes>
          <Route
            path="/"
            element={
              <ProtectedRoute>
                <div>P치gina protegida</div>
              </ProtectedRoute>
            }
          />
          <Route path="/auth" element={<div>P치gina de login</div>} />
        </Routes>
      </MemoryRouter>
    </Provider>,
  );
};

describe('ProtectedRoute Component', () => {
  it('deve renderizar os filhos quando o token estiver presente', () => {
    renderWithRouterAndStore(['/'], {
      userData: { token: 'fake-token' },
    });

    expect(screen.getByText('P치gina protegida')).toBeInTheDocument();
  });

  it('deve redirecionar para /auth quando o token estiver ausente', () => {
    renderWithRouterAndStore(['/'], {
      userData: { token: null },
    });

    expect(screen.getByText('P치gina de login')).toBeInTheDocument();
  });

  it('deve corresponder ao snapshot quando logado', () => {
    const { container } = renderWithRouterAndStore(['/'], {
      userData: { token: 'fake-token' },
    });

    expect(container).toMatchSnapshot();
  });
});
