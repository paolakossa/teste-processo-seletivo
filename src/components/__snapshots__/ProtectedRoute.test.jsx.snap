// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProtectedRoute Component > deve corresponder ao snapshot quando logado 1`] = `
<div>
  <div>
    PÃ¡gina protegida
  </div>
</div>
`;
